### PR TITLE
Changed isone type to AbstractMatrix + added test

### LINKED ISSUE #25516: isone(A) throws error when A is not a StridedMatrix
This is inconsistent (I came across it while looking through Base to see if I can resolve https://github.com/JuliaLang/julia/issues/10385):
```julia
julia> V = view(rand(5,5), [1,3,4], :)
3×5 view(::Array{Float64,2}, [1, 3, 4], :) with eltype Float64:
 0.68645    0.906707  0.474079  0.156336  0.689158 
 0.0366613  0.705923  0.286435  0.169441  0.0788967
 0.602231   0.19818   0.883198  0.796978  0.416841 

julia> isone(V)
ERROR: DimensionMismatch("multiplicative identity defined only for square matrices")
Stacktrace:
 [1] _one at ./array.jl:393 [inlined]
 [2] one at ./array.jl:397 [inlined]
 [3] isone(::SubArray{Float64,2,Array{Float64,2},Tuple{Array{Int64,1},Base.Slice{Base.OneTo{Int64}}},false}) at ./number.jl:45
 [4] top-level scope

julia> isone(rand(3,5))
false
```
A simple fix is to replace `isone(A::StridedMatrix)` definition with `isone(A::AbstractMatrix)`. 

(I don't see any reason why the definitions for `isone` were specialized to `StridedMatrix`.)

### PR DESCRIPTION
**Pull request by ThatcherT**
This is my first contribution! I've made an effort to follow the instructions given in CONTRIBUTING.md. I added a test to show what was breaking before this fix.

Fixes #25516.

### CODE REVIEW

**Comment by dkarrasch** [comment]
Thanks @ThatcherT, and welcome to the Julia community. Could you please add another test that returns `true` (when it should) for a non-strided array?

**Comment by ThatcherT** [comment]
Thanks for the feedback! I will implement. @dkarrasch

**Comment by ThatcherT** [comment]
@dkarrasch I added a positive test for a non-strided array.

File: test/numbers.jl [hunk 1]
```diff
@@ -2641,6 +2641,8 @@ end
     @test !isone(triu(fill(1, 5, 5)))
     @test !isone(zeros(Int, 5, 5))
     @test isone(Matrix(1I, 5, 5))
+    @test !isone(view(rand(5,5), [1,3,4], :))
+    @test isone(view(Diagonal([1,1, 1]), [1,2], 1:2))
```
Comment (dkarrasch) [inline test/numbers.jl:2645]: It appears that this was working before, but it was doing effectively `x == one(x)`, which created a whole dense copy in memory. So the new feature with this PR is that in most cases (except for things like `AbstractQ` from the `qr`s) the `isone`-check is now allocation-free. So we should actually test that as well:
```suggestion
    Dv = view(Diagonal([1,1, 1]), [1,2], 1:2)
    @test isone(Dv)
    @test (@allocated isone(Dv)) == 0
```
The last test fails currently.

**Comment by ThatcherT** [comment]
I've implemented the changes you suggested, including asserting `isone` is now allocation-free

**Comment by dkarrasch** [comment]
Sorry for the delay. Thank you for your contribution!

**Pull request merged**